src/reducers/nft/index.js

```diff
@@ -66,7 +66,8 @@ const fetchNFTsByContractName = createAsyncThunk(
             base_uri: contractMetadata.base_uri,
             fromIndex: selectOwnedTokensForAccountForContract(getState(), { accountId, contractName }).length
         });
-        await dispatch(addTokensMetadata({ accountId, contractName, tokens: tokenMetadata }));
+        const { payload } = await dispatch(addTokensMetadata({ accountId, contractName, tokens: tokenMetadata }));
```

Comment:
This is a code smell; our state should be updated by actions dispatched here rather than splitting it apart with some being tracked in hashes in localState and the rest being tracked in our the redux state.

I'm trying to think of a 'rule-of-thumb' for approaches on implementing these types of patterns.  I think the rule should be "If the state you are managing is stored in Redux, then the loading and error states for that state should also be in Redux.  If the state you are managing is not in Redux, then loading and error states for that state can be tracked in the same components where you are managing the state".  What do you think, does that make sense?